pkg/volume/util/metrics.go

```diff
@@ -56,11 +61,14 @@ func OperationCompleteHook(plugin, operationName string) func(*error) {
 	opComplete := func(err *error) {
 		timeTaken := time.Since(requestTime).Seconds()
 		// Create metric with operation name and plugin name
+		status := statusSuccess
 		if *err != nil {
+			// TODO: Establish well-known error codes to be able to distinguish
+			// user configuration errors from system errors.
+			status = statusFailUnknown
 			storageOperationErrorMetric.WithLabelValues(plugin, operationName).Inc()
-		} else {
-			storageOperationMetric.WithLabelValues(plugin, operationName).Observe(timeTaken)
 		}
+		storageOperationMetric.WithLabelValues(plugin, operationName, status).Observe(timeTaken)
```

Comment:
Will this cause existing dashboards to break because now suddenly this metric will include errors as well. Do we need to capture this in a changelog or something?

Reply:
I have a release note with an action required.

Dashboards shouldn't break however now that the metrics are capturing more cases than before, the semantics of the values may change.  So any alerts based on this metric may need to be adjusted.

Reply:
After some further thought, I'm going to leave the latency histogram as is (only counting successes), and adding a new "status_count" metric to track  number of success and errors